### PR TITLE
Print policy name in schema validation error

### PR DESCRIPTION
**Pull request by andrewalexander**
Fixes https://github.com/capitalone/cloud-custodian/issues/386

### CODE REVIEW

File: c7n/schema.py [hunk 1]
```diff
@@ -55,7 +55,9 @@ def validate(data):
                     ", ".join(dupes)))]
         return []
     try:
-        return [specific_error(errors[0])]
+        resp = specific_error(errors[0])
+        name = isinstance(errors[0].instance, dict) and errors[0].instance.get('name', '')
```
Comment (kapilt) [inline c7n/schema.py]: what do you think about this being unknown instead of empty string.

Comment (andrewalexander) [inline c7n/schema.py]: Makes sense.

**Comment by andrewalexander** [comment]
Updated to return string `'unknown'` if it couldn't find the name or if there wasn't a specific error to be found.

**Comment by kapilt** [comment]
i realized there's still one issue here, in that the fallback mechanism won't work which lead to an index error if specific_error fails.

**Pull request merged**